File: python/src/modules/ingredients_list_provider.py
import ingredient_list_loader as ill

supported_languages = ill.get_languages()
ingredients_language = supported_languages[0]

def get_ingredients_language():
  '''Return the currently selected ingredients language.'''
  return ingredients_language

def validate_ingredients_language(value):
  '''Validates the ingredients language.'''
  if (not(isinstance(value, str)) or value == None or len(value.replace(" ", "")) == 0 or len(value.replace(" ", "")) != 2):
    return 'Language must be a two-letter code (ISO 639-1)'
  if(value.lower() not in supported_languages):
    return 'Language not supported'
  return None

def set_ingredients_language(value):
  '''
    Sets the ingredients language.
    value = two-letter lang code (ISO 639-1).
  '''
  validation_error = validate_ingredients_language(value)
  if(validation_error != None):
    raise Exception("validation_error")
  ingredients_language = value

def get_can_be_vegan_list():
  '''Return the can-be-vegan list of ingredients.'''
  return ill.get_ingredients_lists()[ingredients_language][0]

def get_non_vegan_list():
  '''Return the non-vegan list of ingredients.'''
  return ill.get_ingredients_lists()[ingredients_language][1]

File: src/modules/IngredientsListProvider.ts
import { getLanguages, getIngredientsLists } from './IngredientsListsLoader';

const supportedLanguages = getLanguages();
let ingredientsLanguage = supportedLanguages[0];

/**
 * This functions returns the currently selected ingredients language
 * @return The two-letter code (ISO 639-1) of the currently selected ingredients language
 */
export function getIngredientsLanguage (): string {
  return ingredientsLanguage;
}

/**
 * This functions sets the ingredients language
 * @param value - The two-letter code (ISO 639-1) of the ingredients language to set
 * @throws Will throw an error if validation fails
 */
export function setIngredientsLanguage (value: string | null | undefined): void {
  const validationError = validateIngredientsLanguage(value);

  if (validationError !== null) throw new Error(validationError);

  ingredientsLanguage = value as string;
}

/**
 * This functions returns the can-be-vegan list of ingredients
 * @return The can-be-vegan list of ingredients
 */
export function getCanBeVeganList (): string[] {
  return getIngredientsLists()[ingredientsLanguage][0];
}

/**
 * This functions returns the non-vegan list of ingredients
 * @return The non-vegan list of ingredients
 */
export function getNonVeganList (): string[] {
  return getIngredientsLists()[ingredientsLanguage][1];
}

/**
 * This functions validates the ingredients language
 * @param value - The ingredients language to validate
 * @return A human readable validation error if the ingredients language is not valid. Otherwise null
 */
function validateIngredientsLanguage (value: any): string | null {
  if (
    typeof value !== 'string' ||
    value === null ||
    value.trim().length === 0 ||
    value.length !== 2
  ) { return 'Language must be a two-letter code (ISO 639-1)'; }

  if (!supportedLanguages.includes(value)) { return `Language '${value}' is currently not supported`; }

  return null;
}
